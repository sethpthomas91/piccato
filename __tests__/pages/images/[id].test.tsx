import ImageDetail, { getServerSideProps } from "../../../pages/images/[id]";
import renderer from 'react-test-renderer';
import type { NextApiRequest, NextApiResponse } from 'next';
import { createMocks, RequestMethod } from 'node-mocks-http';

describe ('ImageDetail', () => {
  const mockImageProps = {
      name: "This does not matter",
      url: "some website",
      createdAt: "Some time",
      id: 3
    };

  it('renders correctly', () => {
    const tree = renderer
      .create(<ImageDetail {...mockImageProps}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  const setupFetchStub = (data) => {
    return function fetchStub(_url) {
      return new Promise((resolve) => {
        resolve({
          json: () =>
            Promise.resolve({
              data,
            }),
        })
      })
    }
  }

  it('makes an http call when getServerSideProps is called', async () => {
    const fakeData = { data: "Test"}
    global.fetch = jest.fn().mockImplementation(setupFetchStub(fakeData))

    function mockRequestResponse(method: RequestMethod = 'GET') {
      const {
        req,
        res,
      }: { req: NextApiRequest; res: NextApiResponse } = createMocks({ method });
      req.headers = {
        'Content-Type': 'application/json',
      };
      req.query = {id: "5"}
      return { req, res };
    }
    
    const { req, res } = mockRequestResponse();
    const query = {};
    const resolvedUrl = "Does not matter"
    const params = { 
      id: 1,
      createdAt: "Does not matter",
      name: 'photo name does not matter',
      url: "www.doesnotmatter.com"
    }

    const response = await getServerSideProps({req, res, query, resolvedUrl, params});
    expect(global.fetch).toHaveBeenCalled();
  })
})