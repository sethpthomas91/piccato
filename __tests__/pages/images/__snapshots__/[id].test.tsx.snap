// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDetail renders correctly 1`] = `
<div
  className="container"
>
  <div>
    <img
      height="256"
      src="some website"
      width="256"
    />
  </div>
  <div
    className="detailContainer"
  >
    <h2>
      Name: 
      This does not matter
    </h2>
    <h2>
      Location: 
      some website
    </h2>
  </div>
</div>
`;
